python/cudf/cudf/core/column/string.py

```diff
@@ -4739,7 +4739,11 @@ def _string_column_binop(lhs, rhs, op, out_dtype):
     return out
 
 
-def _get_cols_list(others):
+def _get_cols_list(parent_obj, others):
+
+    parent_index = (
+        parent_obj.index if isinstance(parent_obj, cudf.Series) else parent_obj
+    )
```

Comment:
What if `parent_obj` is `None`, basically the situation where someone is calling functions directly against the `StringColumn` (future StringArray).

Reply:
We currently have this code-flow where we do this: https://github.com/rapidsai/cudf/blob/branch-0.15/python/cudf/cudf/core/column/string.py#L4690

Here we are passing a `StringColumn` so we won't have issue, but incase we get a `Series` and `parent_index` will be `None` and `Index.equals` will return False we hit `Series.reindex` where we donot reindex when the new index is `None`.

Reply:
Should we probably add None check for `parent_index`? Like:

```python
        cols_list = [
            column.as_column(frame.reindex(parent_index), dtype="str")
            if (
                parent_obj is not None
                and isinstance(frame, cudf.Series)
                and not frame.index.equals(parent_index)
            )
            else column.as_column(frame, dtype="str")
            for frame in others
        ]
```

Reply:
LGTM

Reply:
Done